Laravel Valet autocomplete

floq-design opened the pull request:
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
NEW laravel valet autocomplete

**What is the current behavior? (You can also link to an open issue here)**
N/A

**What is the new behavior (if this is a feature change)?**
New autocomplete

**Additional info:**
Autocomplete spec for laravel (v8) valet (v2.15.x)

QuiiBz commented:
Hello, could you change the base branch to `master`? Looks like the current one is your Wordpress branch

floq-design commented:
> Hello, could you change the base branch to `master`? Looks like the current one is your Wordpress branch

TBH I should've branched master before I worked on the wp-cli... So I branched master for this valet work and of course I get all the master commits.. I deleted the `wp.ts` file from this branch so just the one file.. 
If I merge back into master then there will be both `wp.ts` & `valet.ts` which happy to do but didn't want to confuse more than already have!

Lemme know..

mattschrage commented on dev/valet.ts:
```diff
@@ -0,0 +1,718 @@
+        {
+          name: "--authtoken",
+          description: "ngrok.com authtoken identifying a user",
+        },
```
Does this take the auth token as an argument?

floq-design replied:
Yes it does.. added arg

mattschrage commented on dev/valet.ts:
```diff
@@ -0,0 +1,718 @@
+        },
+        {
+          name: "--config",
+          description: "path to config files; they are merged if multiple",
```
Seems like this takes a `path` as an argument.

```
args: {
name: "path",
template: "filepath"
}
```

floq-design replied:
Good spot.. suggestion added

mattschrage commented on dev/valet.ts:
```diff
@@ -0,0 +1,718 @@
+          name: "--host-header",
+          description:
+            "set Host header; if 'rewrite' use local address hostname",
+        },
```
Seems like `--host-head` takes an argument?

floq-design replied:
Yes it does.. args with suggestions added

mattschrage reviewed (changes requested):
Thanks so much for contributing @floq-design! 🙌

I could be wrong as I'm not familiar with the `valet` CLI, but it seems like several options are missing arguments.

(You don't need to provide generators or suggestions for them, but if the option doesn't have a corresponding `arg` object then the completions we generate at runtime will be incorrect!)

Otherwise, great work! Excited to get this merged in :)

QuiiBz commented:
Hello @floq-design, I rebased your branch to remove the previous commits from `wp`. Do you want to resolve Matt's review?

floq-design commented:
> I could be wrong as I'm not familiar with the `valet` CLI...

I'm not familiar with the `share` command for Valet.
These options I believe are based on `ngrok` which I'm also unfamiliar with.
https://ngrok.com/docs#tunnel-definitions

I've reworked a lot of the options based upon the ngrok docs and what the `valet share -h` command shows.
Also added your suggestions - thanks.

floq-design commented:
Sorry made a right hash of pulling latest build into branch and merging.

QuiiBz commented:
Are the commits from wp intentional?

QuiiBz commented:
I again removed the +200 commits. Please pay attention when you pull/merge branches.

QuiiBz reviewed (approved):
Laravel users are going to love this one!
